roaring/roaring.go

```diff
@@ -617,18 +617,34 @@ func (b *Bitmap) UnmarshalBinary(data []byte) error {
 
 	// Read key count in bytes sizeof(cookie):(sizeof(cookie)+sizeof(uint32)).
 	keyN := binary.LittleEndian.Uint32(data[4:8])
-	b.keys = make([]uint64, keyN)
-	b.containers = make([]*container, keyN)
+
+	if len(b.keys) == 0 {
+		b.keys = make([]uint64, 0, keyN)
+		b.containers = make([]*container, 0, keyN)
+	} else if int(keyN) < len(b.keys) { //shrink
+		b.keys = b.keys[:keyN]
+		b.containers = b.containers[:keyN]
+	}
```

Comment:
The containers in `b.containers[keyN:]` should be `nil`'d out before shrinking. Otherwise they won't get GC'd.